### PR TITLE
fix: delete version in hosts urls when a version is published

### PR DESCRIPTION
**Pull request by JYisus**
## Description

- Delete version from urls
- Handle that only one version can be published per product

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [x] I have created tests for my code changes, and the tests are passing.
- [x] I have executed the pre-commit hooks locally.
- [ ] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: engine/admin-api/domain/entity/product_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	publishedVersion := "test-version"
+	product := testhelpers.NewProductBuilder().Build()
+
+	product.UpdatePublishedVersion(publishedVersion)
+
+	assert.Equal(t, publishedVersion, *product.PublishedVersion)
+}
+
+func TestProduct_RemovePublishedVersion(t *testing.T) {
+	product := testhelpers.NewProductBuilder().Build()
```
Comment (alexrodfe) [inline engine/admin-api/domain/entity/product_test.go:37]: this product should be created WithPublishedVersion

**Review by alexrodfe — COMMENTED** [review]
Great done!
Just one tip

File: engine/admin-api/domain/entity/product_test.go (hunk 1 above)
Comment (JYisus) [reply engine/admin-api/domain/entity/product_test.go:37]: True, I added the WithPublishedVersion after this test

**Review by dfinteligenz — APPROVED** [review]

**Review by dfinteligenz — CHANGES_REQUESTED** [review]
There are some tests failing, check them out @JYisus

**Review by kafkaphoenix — DISMISSED** [review]

**Review by alexrodfe — APPROVED** [review]

**Review by kafkaphoenix — APPROVED** [review]

**Review by alexrodfe — APPROVED** [review]

**Review by dfinteligenz — APPROVED** [review]

**Pull request merged**